test/spec/openapi/schema.js

```diff
@@ -849,3 +849,73 @@ test('support query serialization params', async t => {
   t.equal(api.paths['/'].get.parameters[0].style, 'deepObject')
   t.equal(api.paths['/'].get.parameters[0].explode, false)
 })
+
+test('support hide schema', async t => {
+  t.plan(2)
+  const fastify = Fastify()
+  await fastify.register(fastifySwagger, { ...openapiOption, hideSchemas: ['schema-02'] })
+
+  fastify.addSchema({
+    $id: 'schema-01',
+    type: 'object',
+    properties: {
+      id: { type: 'number', description: 'data id' }
+    },
+    required: ['id']
+  })
+
+  fastify.addSchema({
+    $id: 'schema-02',
+    type: 'object',
+    properties: {
+      id: { type: 'number', description: 'data id' }
+    },
+    required: ['id']
+  })
+
+  fastify.addSchema({
+    $id: 'schema-03',
+    type: 'object',
+    properties: {
+      id: { type: 'number', description: 'data id' }
+    },
+    required: ['id']
+  })
+
+  fastify.get('/', {
+    schema: {
+      querystring: { $ref: 'schema-02' },
+      response: {
+        200: {
+          type: 'object'
+        }
+      }
+    }
+  }, () => {})
+
+  await fastify.ready()
+
+  const openapiObject = fastify.swagger()
+  await Swagger.validate(openapiObject)
```

Comment:
I don't understand how this would be valid.

Reply:
`Swagger.validate` only validate `structure`. It does not validate the linkage.
Yes, if `schema-02` is hidden, it should not be a valid one.

Reply:
I think that difference between json schema store and swagger document definition. Validation need every schema but swagger document don't have to display every definitions.

Also, I can move the logic of hiding the schema to the last time of the document building like OpenAPI v2.